src/Analyser/Scope.php

```diff
@@ -485,6 +485,10 @@ private function resolveType(Expr $node): Type
 						return new StaticType($this->getClassReflection()->getName());
 					} elseif ($node->class->parts[0] === 'self') {
 						return new ObjectType($this->getClassReflection()->getName());
+					} elseif ($node->class->parts[0] === 'parent') {
+						if ($this->getClassReflection()->getParentClass() !== false) {
+							return new ObjectType($this->getClassReflection()->getParentClass()->getName());
```

Comment:
This would be nice with a test. It's enough to add it to `NodeScopeResolverTest::dataTypeFromMethodPhpDocs()`.